src/transformers/modeling_tf_utils.py

```diff
@@ -460,6 +458,8 @@ def input_processing(func, config, input_ids, **kwargs):
     has_kwargs = bool(signature.pop("kwargs", None))
     signature.pop("self", None)
     parameter_names = list(signature.keys())
+    main_input_name = parameter_names[0]
```

Comment:
Not needed to resolve the issue but for future we might want to think about cases when there's more than one primary input. For example, [in CLIP the model](https://github.com/huggingface/transformers/blob/bc34c211912697f0cf65fdb63b16e51d2b4aea1c/src/transformers/models/clip/modeling_tf_clip.py#L1344) takes in both `pixel_values` and `input_ids`.

Reply:
The case here is not so much being a primary input, but rather being the first -- we want to support calls like `model(some_dictionary_or_list_of_tf_tensors)`, which is expected to work in Keras. Our `train_step`, which is a modification of the original train step, also uses it ([here](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_tf_utils.py#L1414)).

Would it help to clarify this point if `main_input` was renamed into (e.g.) `first input`? Or maybe the old docstring is in a dire need of an update? 😅

Reply:
Ah, OK, gotcha. `first_input` might be a clearer name but I don't feel strongly so up to you if you want to change :)